apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/ChooseAvailableDaysScreen.tsx

```diff
@@ -0,0 +1,135 @@
+import {StyleSheet} from 'react-native'
+import {DateTime} from 'luxon'
+import {Stack} from 'tamagui'
+import {type TradeChecklistStackParamsList} from '../../../../../navigationTypes'
+import {useAtom, useSetAtom} from 'jotai'
+import {useCallback, useMemo} from 'react'
+import addToSortedArray from '../../../../../utils/addToSortedArray'
+import {UnixMilliseconds} from '@vexl-next/domain/dist/utility/UnixMilliseconds.brand'
+import {
+  availableDateTimesAtom,
+  setDateAndTimeScreenStateActionAtom,
+} from '../../../atoms'
+import {type AvailableDateTimeOption} from '../../../domain'
+import {MINIMUM_AVAILABLE_DAYS_THRESHOLD} from '../../../utils'
+import {useOnFocusAndAppState} from '../../../../ContactListSelect/utils'
+
+export const reactNativeCalendarsDateFormat = 'yyyy-MM-dd'
+
+const styles = StyleSheet.create({
+  header: {
+    marginBottom: 12,
+  },
+})
+
+const calendarTheme: Theme = {
+  calendarBackground: 'transparent',
+  dayTextColor: '#FFFFFF',
+  arrowColor: '#FFFFFF',
+  monthTextColor: '#FFFFFF',
+  selectedDayTextColor: '#FCCD6C',
+  textDayFontWeight: '500',
+  textMonthFontWeight: '500',
+  todayTextColor: '#FFFFFF',
+  agendaDayTextColor: '#FFFFFF',
+  textSectionTitleColor: '#FFFFFF',
+  textDayHeaderFontSize: 14,
+  textDayFontSize: 14,
+  selectedDotColor: 'transparent',
+  selectedDayBackgroundColor: 'transparent',
+  textDisabledColor: '#A0A0AB',
+}
+const today = DateTime.now().toFormat(reactNativeCalendarsDateFormat)
+
+function ChooseAvailableDaysScreen(): JSX.Element {
+  const {t} = useTranslation()
+  const setDateAndTimeScreenState = useSetAtom(
+    setDateAndTimeScreenStateActionAtom
+  )
+  const navigation: NavigationProp<TradeChecklistStackParamsList> =
+    useNavigation()
+  const [availableDateTimes, setAvailableDateTimes] = useAtom(
+    availableDateTimesAtom
+  )
+
+  function handleAvailableDaysChange(day: DateData): void {
+    const millis = UnixMilliseconds.parse(
+      DateTime.fromMillis(day.timestamp).startOf('day').toMillis()
+    )
+
+    if (availableDateTimes.some((dateTime) => dateTime.date === millis)) {
+      setAvailableDateTimes(
+        availableDateTimes.filter(
+          (availableDateTime) => availableDateTime.date !== millis
+        )
+      )
+    } else {
+      setAvailableDateTimes(
+        addToSortedArray(
+          availableDateTimes,
+          (t1, t2) => t1.date - t2.date
+        )({from: millis, to: millis, date: millis})
+      )
+    }
+  }
+
+  const markedDates: MarkedDates = useMemo(
+    () =>
+      availableDateTimes.reduce(
+        (result: MarkedDates, dateTime: AvailableDateTimeOption) => {
+          return {
+            ...result,
+            [DateTime.fromMillis(dateTime.date).toFormat(
+              reactNativeCalendarsDateFormat
+            )]: {selected: true},
+          }
+        },
+        {}
+      ),
+    [availableDateTimes]
+  )
+
+  useOnFocusAndAppState(
+    useCallback(() => {
+      setDateAndTimeScreenState()
```

Comment:
Is this ok? Do we want to do this on every focus? What if user needs to leave for calendar app to check when is he free? 

How about doing the sync before navigation on the root tradingChecklist screen?